src/Documents/Document.ts

```diff
@@ -0,0 +1,15 @@
+import {IDocument} from './IDocument';
+
+export class Document implements IDocument {
+  public entityName(): string {
+      return this.constructor.name;
+  }
+
```

Comment:
We probably do not need a class nor interface for `Document` at all. `Raven.Client` (C#) does not have a concept of document. It just returns objects/entities. We could just use `JSON.parse()` on the document body from server's response and `JSON.serialize()` if we need a string representation.